File: stats_series_01_week_05.R
# Snippet 1: Introduction to Sampling
population <- 1:10000
sample_random <- sample(population, 100) # Random Sampling

library(caret)

# Stratified Sampling
y <- ifelse(population <= 5000, 0, 1)
stratified_sample <- createDataPartition(y, times = 1, p = 0.1, list = FALSE)

# Snippet 2: Sampling Distribution & Central Limit Theorem
means <- replicate(1000, mean(sample(population, 30))) # Sampling Distribution
hist(means, breaks=30)

# Snippet 3: Estimation: Point and Interval
confidence_level <- 0.95
mean <- mean(sample_random)
standard_error <- sd(sample_random) / sqrt(length(sample_random))
confidence_interval <- qt(confidence_level, df=length(sample_random)-1) * standard_error
CI_lower <- mean - confidence_interval
CI_upper <- mean + confidence_interval

File: stats_series_01_week_05.py
# Snippet 1: Introduction to Sampling
import random

population = range(1, 10001)
sample_random = random.sample(population, 100) # Random Sampling

from sklearn.model_selection import train_test_split

# Stratified Sampling
X, y = population, [0 if i <= 5000 else 1 for i in population]
X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y, test_size=0.1)

# Snippet 2: Sampling Distribution & Central Limit Theorem
import numpy as np

means = [np.mean(random.sample(population, 30)) for _ in range(1000)] # Sampling Distribution
plt.hist(means, bins=30)
plt.show()

# Snippet 3: Estimation: Point and Interval
import scipy.stats as stats

confidence_level = 0.95
degrees_freedom = len(sample_random) - 1
mean = np.mean(sample_random)
standard_error = stats.sem(sample_random)
confidence_interval = stats.t.interval(confidence_level, degrees_freedom, mean, standard_error)
